pkg/preparer/podprocess/finish_output.go

```diff
@@ -0,0 +1,48 @@
+package podprocess
+
+import (
+	"database/sql"
+
+	"github.com/square/p2/pkg/launch"
+	"github.com/square/p2/pkg/types"
+	"github.com/square/p2/pkg/util"
+
+	_ "github.com/mattn/go-sqlite3"
+)
+
+// Represents a row in the sqlite database indicating the exit of a runit process.
+type FinishOutput struct {
+	PodID        types.PodID         `json:"pod_id"`
+	LaunchableID launch.LaunchableID `json:"launchable_id"`
+	EntryPoint   string              `json:"entry_point"`
+	PodUniqueKey types.PodUniqueKey  `json:"pod_unique_key"`
+
+	// These two model the arguments given to the ./finish script under runit:
+	// (http://smarden.org/runit/runsv.8.html)
+	ExitCode   int `json:"exit_code"`
+	ExitStatus int `json:"exit_status"`
+}
+
+func (f FinishOutput) Insert(db *sql.DB) error {
```

Comment:
SQL DB style recommendation - rather than putting this function on the data type, put it on a service type that has a function `InsertFinish(finishOutput, db)`. Constraining DB operations to a single interface will make it easier to stub later. 

It also results in the various queries / statements and sqlite implementation being consolidated to a single location.